### DISCUSSION TITLE
subscription related details

### DISCUSSION THREAD

**Discussion by shashikantpawar21** [Eventuous/eventuous#132] [Q&A]
services.AddSubscription<AllStreamSubscription, AllStreamSubscriptionOptions>(
        "**MainProjection**",
        builder => builder
            ****.Configure(cfg => cfg.ConcurrencyLimit = 2)**
            .UseCheckpointStore<MongoCheckpointStore>()**
            .AddEventHandler<**SecondaryProjection**>()
            .AddEventHandler<**ThirdProjection**>()
            .AddEventHandler<**FourthProjection**>()
            **.WithPartitioningByStream(2));**

What is the use of bolded line in above syntax .. Does they help with something ??
ConcurrencyLimit
UseCheckpointStore
WithPartitioningByStream

**Comment by JulianMay** [comment]
Please read the documentation : https://eventuous.dev/docs/subscriptions/eventstore/all-stream-sub/#concurrent-event-handlers

**Reply by shashikantpawar21** [reply]
what I understand from the articles is 

concurrency limit is that multiple added EventHandler would listen to the subscription in parallelly 
And partition is about mentioning about a block not would run sequentially to avoid issue 

Not sure in our application, I have set the things as above .. but looks like in some rare cases it is calling the event handler twice for one event mostly the new one .. 

Also, i am not seeing much mongo projection logging happening as compared to other parts

**Reply by devployment** [reply]
Link to the docs changed

https://eventuous.dev/docs/infra/esdb/#concurrent-event-handlers-1

**Reply by alexeyzimarev** [reply]
Checkpoint store, same page https://eventuous.dev/docs/infra/esdb/#checkpoint-store

The concurrency limit configuration was a mistake in the sample app, it's now removed; only the partitioner is still there as a demo. You don't need those things if you don't have a lot of events. The checkpoint can be configured per subscription, or globally. It's all in the docs.
